### DISCUSSION TITLE
What's still missing?

### DISCUSSION THREAD

**Discussion by manchicken** [XAMPPRocky/octocrab#255] [Ideas]
@XAMPPRocky, or anybody else, what's missing that you'd like to see added? I'm having fun contributing to this repo and I'd like to do some more.

**Comment by XAMPPRocky** [comment]
Thank you for your interest! Right now the biggest thing that needs to be done is #168 which will remove the dependency on `reqwest`. There's also adding `IntoFuture` implementations for each of the builders to replace the `send` method. There's probably also a bunch of missing GitHub APIs.

**Reply by manchicken** [reply]
So for the `reqwest` removal, what is the vision for how you'd want to refactor the HTTP client? Would that be dependency injection, or would you want to just shift to a different client?

**Reply by XAMPPRocky** [reply]
@manchicken Check out the `kube` crate for how it accepts http clients. We’d want the same approach.

**Reply by manchicken** [reply]
Ok, so you're just talking about swapping `reqwest` out for https://crates.io/crates/http ?

**Reply by XAMPPRocky** [reply]
Yes

**Reply by manchicken** [reply]
Ok, I'm looking at all of this, and on the issue #99 you suggest moving one endpoint at a time. I'm wondering how you'd recommend doing that given the shared code.

Would you see each of the endpoints taking ownership of the transport layer that they use, or would you see the development of an alternate shared transport layer and just migrate one endpoint at a time?
